File: employee-portal/src/services/operationsService.ts
import { employeeApiClient } from "./apiClient";
import type {
  CashDepositHistoryEntry,
  CashDepositRequest,
  CashDepositResponse,
  CustomerSearchResponse,
} from "@/types/operations";

/** The "Employee Operations API" — see docs/architecture/employee-operations.md. */
export const operationsService = {
  async customerSearchByAccountNumber(accountNumber: string): Promise<CustomerSearchResponse> {
    const { data } = await employeeApiClient.get<CustomerSearchResponse>("/api/v1/operations/customer-search", {
      params: { accountNumber },
    });
    return data;
  },

  async customerSearchByCustomerId(customerId: string): Promise<CustomerSearchResponse> {
    const { data } = await employeeApiClient.get<CustomerSearchResponse>("/api/v1/operations/customer-search", {
      params: { customerId },
    });
    return data;
  },

  async cashDeposit(request: CashDepositRequest): Promise<CashDepositResponse> {
    const { data } = await employeeApiClient.post<CashDepositResponse>("/api/v1/operations/cash-deposits", request);
    return data;
  },

  async cashDepositHistory(): Promise<CashDepositHistoryEntry[]> {
    const { data } = await employeeApiClient.get<CashDepositHistoryEntry[]>("/api/v1/operations/cash-deposits/history");
    return data;
  },
};

File: backend/services/employee-service/src/main/java/com/banksphere/employee/service/OperationsService.java
package com.banksphere.employee.service;

import com.banksphere.employee.dto.CashDepositHistoryEntry;
import com.banksphere.employee.dto.CashDepositRequest;
import com.banksphere.employee.dto.CashDepositResponse;
import com.banksphere.employee.dto.CustomerSearchResponse;

import java.util.List;
import java.util.UUID;

/** The "Employee Operations API" — see docs/architecture/employee-operations.md and ADR-007. */
public interface OperationsService {

    /** Exactly one of {@code accountNumber}/{@code customerId} must be non-null. */
    CustomerSearchResponse customerSearch(String accountNumber, UUID customerId, String bearerToken);

    /** The acting employee's identity/branch come only from {@code actingEmployeeId} (their own verified JWT subject), never the request body. */
    CashDepositResponse cashDeposit(CashDepositRequest request, UUID actingEmployeeId, String bearerToken);

    /** The acting employee's own branch's most recent cash deposits. */
    List<CashDepositHistoryEntry> cashDepositHistory(UUID actingEmployeeId, String bearerToken);
}
